### PR TITLE
Add psalm template annotation to container get method

### PR DESCRIPTION
**Pull request by juliusknorr**
Add psalm annotations in order to properly detect the return type of `\OC::$server->get(MyClass::class)`.

Even though DI should be used, this allows to properly get the return type without manually specifiying it when using `get()`

### CODE REVIEW

**Review by nickvergessen — APPROVED** [review]

**Review by CarlSchwan — APPROVED** [review]

File: lib/private/AppFramework/Utility/SimpleContainer.php [hunk 1]
```diff
@@ -52,6 +52,11 @@ public function __construct() {
 		$this->container = new Container();
 	}
 
+	/**
+	 * @template T
+	 * @param class-string<T> $id
```
Comment (CarlSchwan) [inline lib/private/AppFramework/Utility/SimpleContainer.php:57]:
```suggestion
     * @psalm-suppress MoreSpecificImplementedParamType
	 * @param class-string<T> $id
```

This should hopefully fix the issue :)

**Comment by CarlSchwan** [comment]
As I recently discovered, it seems like the `Static code analysis / static-code-analysis-ocp` and `Static code analysis / static-code-analysis` are not run correctly anymore and don't report errors anymore, so we should refrain to merge this until we figure out what is wrong (I'm on it)

**Comment by CarlSchwan** [comment]
I added a commit that fixes the new issues reported by psalm. Please check it :)

File: lib/private/Server.php [hunk 2]
```diff
@@ -1156,8 +1159,9 @@ public function __construct($webRoot, \OC\Config $config) {
 		/** @deprecated 19.0.0 */
 		$this->registerDeprecatedAlias('CommentsManager', ICommentsManager::class);
 
-		$this->registerAlias(\OC_Defaults::class, 'ThemingDefaults');
-		$this->registerService('ThemingDefaults', function (Server $c) {
+		$this->registerAlias(\OC_Defaults::class, ThemingDefaults::class);
+		$this->registerDeprecatedAlias('ThemingDefaults', ThemingDefaults::class);
```
Comment (nickvergessen) [inline lib/private/Server.php:1163]:
```suggestion
		/** @deprecated 24.0.0 */
		$this->registerDeprecatedAlias('ThemingDefaults', ThemingDefaults::class);
```

**Review by nickvergessen — APPROVED** [review]

**Comment by juliushaertl** [comment]
Obsolete with https://github.com/nextcloud/server/pull/33388

**Pull request closed without merging**